components/dashboard/src/components/podkit/buttons/LinkButton.tsx

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { Link } from "react-router-dom";
+import { Button, ButtonProps } from "@podkit/buttons/Button";
+import React from "react";
+
+export interface LinkButtonProps extends ButtonProps {
+    asChild?: false;
+    href: string;
+}
+
+/**
+ * A HTML anchor element styled as a button.
+ */
+export const LinkButton = React.forwardRef<HTMLButtonElement, LinkButtonProps>(
+    ({ className, variant, size, asChild, children, href, ...props }, ref) => {
+        return (
+            <Button className="transition-width" asChild {...props} ref={ref}>
```

Comment:
@selfcontained I have still not been able to figure out how to animate the width. My intuition is telling me, that although animating the width when you change it via CSS like in a query selector is pretty simple, CSS will not let you animate the changes that are needed in order to fit that content, because it makes sure it can fit it all on the frame the new component is rendered.

Hence, my skepticism about the approach of animating width. I am not sure what some good alternatives would look like (besides maybe the one where we reserve space in the layout for the loading indicator).